### ISSUE TITLE
Flipper is using "Memory" adapter in Rails console with ActiveRecord adapter installed

### ISSUE THREAD

**Issue by adamlogic** [jnunemaker/flipper#610]
In my local Rails console (development), I was getting very confused why my feature flags were not being persisted. It looks like Flipper is using the memory adapter even though I have the ActiveRecord adapter installed.

```ruby
> Flipper.adapter
=> #<Flipper::Adapters::Memory:6940 name=:memory, source={}>

> Flipper.configuration.adapter
=> #<Flipper::Adapters::ActiveRecord:0x000000010e0efbf0 ...
```

I have the following gems installed:

```ruby
gem "flipper"
gem "flipper-active_record"
gem "flipper-ui"
```

And I've setup Flipper UI with Rack authentication [as documented](https://www.flippercloud.io/docs/ui#basic-authentication-via-rack):

```ruby
flipper_app = Flipper::UI.app(Flipper.instance) do |builder|
  builder.use Rack::Auth::Basic do |username, password|
    username == "user" && password == "pass"
  end
end
mount flipper_app, at: "/flipper"
```

I think I've traced this to the fact that routes are loaded before the Flipper-ActiveRecord initializer, so calling `Flipper.instance` in `routes.rb` is [memoizing the default adapter](https://github.com/jnunemaker/flipper/blob/master/lib/flipper.rb#L46) via `Thread.current[:flipper_instance]`.

Clearing out `Flipper.instance` resolves the issue:

```ruby
> Flipper.adapter
=> #<Flipper::Adapters::Memory:6940 name=:memory, source={}>

> Flipper.instance = nil
=> nil

> Flipper.adapter
=> #<Flipper::Adapters::ActiveRecord:0x000000010e0efbf0 ...
```

Here's a [sample repo](https://github.com/adamlogic/flipper-test) that reproduces the issue. Just run `rails c` and `Flipper.adapter`, and you should see the same result.

I have not confirmed if this is only an issue with Rails console and if it's only an issue in development. I was seeing some unexpected behavior in the browser as well, but now it seems to be working, so I'm not quite sure.

Am I doing something unusual here to be the first to come across this issue?

**Comment by adamlogic** [comment]
Looks like with this change everything works as expected (so far):

```diff
- flipper_app = Flipper::UI.app(Flipper.instance) do |builder|
+ flipper_app = Flipper::UI.app(Flipper) do |builder|
```

Perhaps it's just a documentation issue?

**Comment by jnunemaker** [comment]
@adamlogic hey! Appreciate you writing this up so thoroughly. You are correct on all counts. This looks like a docs issue. If you just remove Flipper.instance entirely it should work like so: 

```ruby
flipper_app = Flipper::UI.app do |builder|
```

I tested in your app both ways and I see how it is in fact being set to memory. I'm updating the docs now. I typically use a constraint so I haven't hit this issue and I don't think I wrote those docs. 

I'd like to know what @bkeepers thinks too. Should we reset `Flipper.instance` somehow when the auto config stuff you wrote happens?
